### DISCUSSION TITLE
Custom Behaviors

### DISCUSSION THREAD

**Discussion by dcolestock** [manna-harbour/miryoku#131] [Q&A] [answered]
Is there a better way to add custom behaviors than editing `miryoku.dtsi`?  It didn't work when I tried putting them in `custom_config.h` or tried putting them into a `.dtsi` file that `custom_config.h` includes.

Also, is there a way to create a new custom behavior from another? Say if I want to make a modified version of `homerow_mods`, but just want to overwrite the `tapping_term_ms` so my picky fingers have a little more time?

**Comment by manna-harbour** [comment, marked as answer]
> Is there a better way to add custom behaviors than editing `miryoku.dtsi`? It didn't work when I tried putting them in `custom_config.h` or tried putting them into a `.dtsi` file that `custom_config.h` includes.

I'll be adding a `custom_keymap.dtsi` for that at some point (https://github.com/manna-harbour/miryoku/issues/94). But for now you'd need to create that yourself and add an include to `miryoku.dtsi`. 

> Also, is there a way to create a new custom behavior from another? Say if I want to make a modified version of `homerow_mods`, but just want to overwrite the `tapping_term_ms` so my picky fingers have a little more time?

I'll also be adding support for that (https://github.com/manna-harbour/miryoku/issues/33), but for now you'd add the new behaviour to `custom_keymap.dtsi` (after adding the include) and customise Base to use the new behaviour on the pinkies (https://github.com/manna-harbour/miryoku/discussions/85). I don't think you can create a custom behaviour from another custom behaviour, you'd probably need to create the new behaviour from `behavior-hold-tap` in the same way as is done now for `hm`.
